connman/src/service.c

```diff
@@ -4601,30 +4650,86 @@ int __connman_service_reset_ipconfig(struct connman_service *service,
 
 	__connman_ipconfig_unref(ipconfig);
 
-	if (type == CONNMAN_IPCONFIG_TYPE_IPV4)
-		service->ipconfig_ipv4 = new_ipconfig;
-	else if (type == CONNMAN_IPCONFIG_TYPE_IPV6)
-		service->ipconfig_ipv6 = new_ipconfig;
+	err = reset_ipconfig(service, ipconfig, new_ipconfig, type, new_method,
+							state, new_state);
 
-	if (is_connecting_state(service, state) ||
-					is_connected_state(service, state))
-		__connman_ipconfig_enable(new_ipconfig);
+	__connman_ipconfig_unref(ipconfig);
```

Comment:
Hm, this may have been a cleanup error. Let me check this.

Reply:
I think the whole
```
if (is_connecting_state(service, state) ||
					is_connected_state(service, state))
		__connman_network_clear_ipconfig(service->network, ipconfig);
__connman_ipconfig_unref(ipconfig);
```
needs to be removed since clearing is done in `reset_ipconfig()` after the method from the config is used.  This part I guess I can test with faked clat so that this part does not alter functionality as it would then, after removal of that part of code, be able to get the real method from the ipconfig.

Reply:
Does not seem to make any difference. Similar D-Bus messaging happens AFAIK. First `settings_changed()' call still holds empty IPv4 as it is not set and then when plugin calls to indicate state after setting everything up the IPv4 info contains necessary bits. Checked with gdb bt that there are no excess calls to that `settings_changed()` after these changes. Resolving this as the unnecessary code was eliminated.